doc/releases/release-notes-3.5.rst

```diff
@@ -367,26 +367,164 @@ Drivers and Sensors
+    * :c:func:`coap_packet_remove_option`
+    * :c:func:`coap_packet_set_path`
+
+* Connection Manager:
+
+  * Added support for auto-connect and auto-down behaviors (controlled by
+    :c:enum:`CONN_MGR_IF_NO_AUTO_CONNECT` and :c:enum:`CONN_MGR_IF_NO_AUTO_DOWN`
+    flags).
+  * Split Connection Manager APIs into separate header files.
+  * Extended Connection Manager documentation to cover new functionalities.
+
+* DHCP:
+
+  * Added support for DHCPv4 unicast replies processing.
+  * Added support for DHCPv6 protocol.
+
+* ICMP:
+
+  * Fixed ICMPv6 error message type check.
+  * Reworked ICMP callback registration and handling, which allows to register
+    multiple handlers for the same ICMP message.
+  * Introduced an API to send ICMP Echo Request (ping).
+  * Added possibility to register offloaded ICMP ping handlers.
+  * Added support for setting packet priority for ping.
+
+* IPv6:
+
+  * Made sure that ongoing DAD procedure is cancelled when IPv6 address is removed.
+  * Fixed a bug, where Solicited-Node multicast address could be removed while
+    still in use.
 
 * LwM2M:
 
   * Added support for tickless mode. This removes the 500 ms timeout from the socket loop
     so the engine does not constantly wake up the CPU. This can be enabled by
     :kconfig:option:`CONFIG_LWM2M_TICKLESS`.
   * Added new :c:macro:`LWM2M_RD_CLIENT_EVENT_DEREGISTER` event.
+  * Block-wise sending now supports LwM2M read and composite-read operations as well.
+    When :kconfig:option:`LWM2M_COAP_BLOCK_TRANSFER` is enabled, any content that is larger than
+    :kconfig:option:`CONFIG_LWM2M_COAP_MAX_MSG_SIZE` is split into a block-wise transfer.
+  * Block-wise transfers don't require tokens to match anymore as this was not in line
+    with CoAP specification (CoAP doesn't require tokens re-use).
+  * Various fixes to bootstrap. Now client ensures that Bootstrap-Finnish command is send,
+    before closing the DTLS pipe. Also allows Bootstrap server to close the DTLS pipe.
+    Added timeout when waiting for bootstrap commands.
+  * Added support for X509 certificates.
+  * Various fixes to string handling. Allow setting string to zero length.
+    Ensure string termination when using string operations on opaque resources.
+  * Added support for Connection Monitoring object version 1.3.
+  * Added protection for Security object to prevent read/writes by the server.
+  * Fixed a possible notification stall in case of observation token change.
+  * Added LwM2M interoperability test-suite against Leshan server.
+  * API documentation improvements.
+  * Several other minor fixes and improvements.
+
+* Misc:
+
+  * Time and timestamps in the network subsystem, PTP and IEEE 802.15.4
+    were more precisely specified and all in-tree call sites updated accordingly.
+    Fields for timed TX and TX/RX timestamps have been consolidated. See
+    :c:type:`net_time_t`, :c:struct:`net_ptp_time`, :c:struct:`ieee802154_config`,
```

Comment:
`net_time_t` does not link to anything as `net_time.h` does not define any doxygen group nor does it appear in the Sphinx documentation. May be worth fixing?

Reply:
TBH I kind of miss an idea where to put such a reference (and least or now) and defining a whole new doc page for a header with a single typedef sounds like a overshot to me.

Reply:
> TBH I kind of miss an idea where to put such a reference (and least or now) and defining a whole new doc page for a header with a single typedef sounds like a overshot to me.

Ya, I feel you. Would it make sense to start creating some kind of "Utilities" group under Connectivity? https://docs.zephyrproject.org/latest/doxygen/html/group__connectivity.html, and then maybe an RST as a subpage of https://docs.zephyrproject.org/latest/connectivity/networking/api/apis.html.

In case this helps identify other candidates for maybe grouping other APIs alongside `net_time_t` (socketutils?). Note that these are also currently not showing in the Sphinx docs so you might want to have a look in any case.

```
❯ grep -L -r '@defgroup\|@addtogroup' include/zephyr/net/
include/zephyr/net/http/parser_url.h
include/zephyr/net/http/parser.h
include/zephyr/net/http/parser_state.h
include/zephyr/net/http/service.h
include/zephyr/net/loopback.h
include/zephyr/net/socket_offload.h
include/zephyr/net/ieee802154_pkt.h
include/zephyr/net/ipv4_autoconf.h
include/zephyr/net/net_time.h
include/zephyr/net/socketutils.h
include/zephyr/net/ieee802154_radio_openthread.h
include/zephyr/net/bt.h
include/zephyr/net/canbus.h
```

Reply:
For now I've just added a new section under "Time Sensitive Networking" for net_time. Other headers likely need a review, but from what I see in many cases they either not use doxygen at all, or are documenting internal APIs.